### PR TITLE
Testing: bug-fix db flags generation when waiting for DB

### LINKED ISSUE #1178: Issue when running unit tests in Docker Container
When trying to run unit tests in Docker using the command `npm run test:php`, I encounter the following error:

Screenshot 2020-12-21 at 18 48 32

This happens repeatedly until the process is killed.

After doing some investigation, I believe it may be related to the change introduced in https://github.com/Automattic/woocommerce-payments/commit/65ab6d77fb57443bc1301f3829ec2521ac82f86b which updated some of the database connection logic in `install-wp-tests.sh`, after reverting to the prior version to this I was able to run the unit tests, but we'd need to investigate to confirm this.

### PR DESCRIPTION
**Pull request by kalessil**
Fixes #1178

#### Changes proposed in this Pull Request

* Replicated DB connection probe flags when waiting for DB

#### Testing instructions

* E2E tests job shall pass

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: bin/install-wp-tests.sh [hunk 1]
```diff
@@ -39,7 +39,23 @@ wp() {
...
+		if [ $(echo $DB_SOCK_OR_PORT | grep -e '^[0-9]\{1,\}$') ]; then
+			EXTRA=" --host=$DB_HOSTNAME --port=$DB_SOCK_OR_PORT --protocol=tcp"
+		elif ! [ -z $DB_SOCK_OR_PORT ] ; then
+			EXTRA=" --socket=$DB_SOCK_OR_PORT"
+		elif ! [ -z $DB_HOSTNAME ] ; then
+			EXTRA=" --host=$DB_HOSTNAME --protocol=tcp"
+		fi
+	fi
+
+	local MYSQLADMIN_FLAGS="--user=$DB_USER --password=$DB_PASS $EXTRA"
```
Comment (luizreis) [inline bin/install-wp-tests.sh:58]: It looks like this code is duplicated here and in `install_db`. What do you think of extracting it to a new function, e.g. `get_db_connection_string`, and using that function in those places?

**Review by luizreis — APPROVED** [review]
Works well! Thanks for submitting this fix. I left a suggestion, but it is non-blocking, so I'm approving the PR.

File: bin/install-wp-tests.sh (hunk 1 above)
Comment (kalessil) [reply bin/install-wp-tests.sh:58]: Gimme a sec to try, I want to move at the very beginning of the script.

**Comment by dmallory42** [comment]
Just to confirm, this resolves the issue in #1178 for me. Thanks for submitting so quickly!

**Review by dmallory42 — APPROVED** [review]

File: bin/install-wp-tests.sh (hunk 1 above)
Comment (kalessil) [reply bin/install-wp-tests.sh]: The incorporated solution is proposed by @luizreis, thank you :+1:

**Pull request merged**